### PR TITLE
bF consolidate Txs

### PR DESCRIPTION
**Pull request by husobee**
### Summary

Consolidation of TXs for bitflyer userdrains.

### Type of change ( select one )

- [ ] Product feature
- [ ] Bug fix
- [ ] Performance improvement
- [ ] Refactor
- [ ] Other

### Tested Environments

- [ ] Development
- [ ] Staging
- [ ] Production

### Before submitting this PR:

- [ ] Does your code build cleanly without any errors or warnings?
- [ ] Have you used auto closing keywords?
- [ ] Have you added tests for new functionality?
- [ ] Have validated query efficiency for new database queries?
- [ ] Have documented new functionality in README or in comments?
- [ ] Have you squashed all intermediate commits?
- [ ] Is there a clear title that explains what the PR does?
- [ ] Have you used intuitive function, variable and other naming?
- [ ] Have you requested security / privacy review if needed
- [ ] Have you performed a self review of this PR?

### CODE REVIEW

File: promotion/datastore.go [hunk 1]
```diff
@@ -686,6 +702,21 @@ func (pg *Postgres) SaveClaimCreds(creds *ClaimCreds) error {
...
+	if err != nil {
+		return err
+	}
+	defer pg.RollbackTx(tx)
+
+	stmt := "update claim_drain set status = 'complete', completed=true, completed_at = now() where batch_id = $1"
+	if _, err := tx.Exec(stmt, batchID); err == nil {
+		return tx.Commit()
+	}
+	return nil
```
Comment (clD11) [inline promotion/datastore.go:717]: would we want to return an error here for tx.Exec?

File: promotion/datastore.go [hunk 2]
```diff
@@ -908,6 +939,91 @@ ORDER BY created_at DESC
...
+			update claim_drain set
+				erred = true,
+				errcode = $1,
+				status = $2
+			where batch_id = $3`, errCode, status, batchID)
+		if err != nil {
+			return attempted, err
+		}
+		// inform sentry about this error
+		sentry.CaptureException(err)
```
Comment (clD11) [inline promotion/datastore.go:1007]: should we enrich the error with some more info like batch id and maybe the job name. Might apply to some of the other errors as well

Comment (husobee) [reply promotion/datastore.go]: https://github.com/brave-intl/bat-go/pull/973/commits/804b092ef331a7fa090c0dd7b48b391e2fb5a1fc addressed here, think the error classification is good for sentry, dont think we want in depth details.

File: promotion/datastore.go (hunk 1 above)
Comment (husobee) [reply promotion/datastore.go]: https://github.com/brave-intl/bat-go/pull/973/commits/804b092ef331a7fa090c0dd7b48b391e2fb5a1fc addressed here.

**Review by clD11 — APPROVED** [review]

**Pull request merged**